### PR TITLE
feat(StorageInsights): Adding samples

### PR DESCRIPTION
**Pull request by vishwarajanand**
Fixes https://github.com/GoogleCloudPlatform/php-docs-samples/issues/1899

Depends on https://github.com/GoogleCloudPlatform/php-docs-samples/pull/1987

Tests pass on prod:
```
➜  storageinsights git:(storage_insights) ✗ ../testing/vendor/bin/phpunit -c phpunit.xml.dist                                                         
PHPUnit 9.6.18 by Sebastian Bergmann and contributors.

Warning:       XDEBUG_MODE=coverage or xdebug.mode=coverage has to be set

.....                                                               5 / 5 (100%)

Time: 00:19.591, Memory: 10.00 MB

OK (5 tests, 7 assertions)
➜  storageinsights git:(storage_insights) ✗
```

I managed to create an insights report: `projects/the-book-club-337319/locations/us-west1/reportConfigs/02f2961c-0e8d-4339-937c-f0bda263447c`

### CODE REVIEW

File: storageinsights/src/create_inventory_report_config.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    string $sourceBucket,
+    string $destinationBucket
+): void {
+    $storageInsightsClient = new StorageInsightsClient();
+
+    $reportConfig = (new ReportConfig())
+        ->setDisplayName('Example inventory report configuration')
+        ->setFrequencyOptions((new FrequencyOptions())
+            ->setFrequency(FrequencyOptions\Frequency::WEEKLY)
+            ->setStartDate((new Date())
```
Comment (saranshdhingra) [inline storageinsights/src/create_inventory_report_config.php:48]: Is this the best way to set the day?
Does the interface accept any other format, which doesn't ask the user to make such verbose calls?

File: storageinsights/src/get_inventory_report_names.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    string $projectId,
+    string $bucketLocation,
+    string $inventoryReportConfigUuid
+): void {
+    $storageInsightsClient = new StorageInsightsClient();
+
+    $reportConfigName = $storageInsightsClient->reportConfigName($projectId, $bucketLocation, $inventoryReportConfigUuid);
+    $reportConfig = $storageInsightsClient->getReportConfig($reportConfigName);
+    $extension = $reportConfig->hasCsvOptions() ? 'csv' : 'parquet';
+    printf('You can use the Google Cloud Storage Client '
```
Comment (saranshdhingra) [inline storageinsights/src/get_inventory_report_names.php:38]: nit: Use `print` instead?

File: storageinsights/src/list_inventory_report_configs.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace Google\Cloud\Samples\StorageInsights;
+
+# [START storageinsights_list_inventory_report_configs]
+use Google\Cloud\StorageInsights\V1\StorageInsightsClient;
+
+/**
+ * Example:
+ * ```
+ * list_inventory_report_configs($projectId, $location);
```
Comment (saranshdhingra) [inline storageinsights/src/list_inventory_report_configs.php:26]: nit: Let's add this call for other samples too?

**Review by saranshdhingra — COMMENTED** [review]
Just noticed that there is an error regarding a PHP extension in the logs, can you check it?

File: storageinsights/src/create_inventory_report_config.php (hunk 1 above)
Comment (vishwarajanand) [reply storageinsights/src/create_inventory_report_config.php:48]: [This code pointer](https://github.com/googleapis/google-cloud-php/blob/ec3c0442772d165bdfecba8eaceb60aa3677d139/StorageInsights/src/V1/FrequencyOptions.php#L116-L119) requires it to be `\Google\Type\Date::class`
So, I think we can't pass `DateTime` or a `string` here.

**Comment by vishwarajanand** [comment]
@saranshdhingra  seems the warning is due to missing extension `pdo_srv` which is not used in this library. I will try to fix that in a separate PR.

**Comment by vishwarajanand** [comment]
@saranshdhingra  I raised a PR to fix the missing PHP extension warning:
https://github.com/googleapis/testing-infra-docker/pull/373

**Review by ajupazhamayil — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**